lib/puppet/type/cloudformation_stack.rb

```diff
@@ -0,0 +1,223 @@
+  end
+
+  newproperty(:outputs) do
+    desc 'A hash of stack outputs. (readonly)'
+    def should_to_s(value)
+      value.inspect
+    end
+    def is_to_s(value)
+      value.inspect
+    end
+  end
+
+  newproperty(:parameters) do
+    desc 'A hash of input parameters.'
+    def insync?(is)
+      is == should
+    end
+    def should_to_s(value)
+      value.inspect
+    end
+    def is_to_s(value)
+      value.inspect
+    end
+  end
+
+  newparam(:policy_body) do
+    desc 'Structure containing the stack policy body. For more information, go to Prevent Updates to Stack Resources in the AWS CloudFormation User Guide. You can specify either the policy_body or the policy_url parameter, but not both.'
+    validate do |value|
+      fail 'policy_body must be a String' unless value.is_a?(String)
+    end
+  end
+
+  newparam(:policy_url) do
+    desc 'Location of a file containing the stack policy. The URL must point to a policy (maximum size: 16 KB) located in an S3 bucket in the same region as the stack. You can specify either the policy_body or the policy_url parameter, but not both.'
+    validate do |value|
+      fail 'policy_url must be a String' unless value.is_a?(String)
+    end
+  end
+
+  validate do
+    fail "You can specify either policy_body or policy_url but not both for the cloudformation stack [#{self[:name]}]" if self[:policy_body] && self[:policy_url]
+  end
+
+  newparam(:resource_types, :array_matching => :all) do
+    desc 'The list of resource types that you have permissions to work with for this stack. (optional)'
+    def insync?(is)
+      is.to_set == should.to_set
+    end
+    validate do |value|
+      fail 'resource_types must be a string.' unless value.is_a?(String)
+    end
+  end
+
+  newproperty(:role_arn) do
+    desc 'The Amazon Resource Name (ARN) of an AWS Identity and Access Management (IAM) role that is associated with the stack.'
+    validate do |value|
+      fail 'role_arn must be a String' unless value.is_a?(String)
+    end
+  end
+
+  newproperty(:status) do
+    desc 'The status of the stack. (readonly)'
+    newvalue(:CREATE_IN_PROGRESS)
+    newvalue(:CREATE_FAILED)
+    newvalue(:CREATE_COMPLETE)
+    newvalue(:ROLLBACK_IN_PROGRESS)
+    newvalue(:ROLLBACK_FAILED)
+    newvalue(:ROLLBACK_COMPLETE)
+    newvalue(:DELETE_IN_PROGRESS)
+    newvalue(:DELETE_FAILED)
+    newvalue(:DELETE_COMPLETE)
+    newvalue(:UPDATE_IN_PROGRESS)
+    newvalue(:UPDATE_COMPLETE_CLEANUP_IN_PROGRESS)
+    newvalue(:UPDATE_COMPLETE)
+    newvalue(:UPDATE_ROLLBACK_IN_PROGRESS)
+    newvalue(:UPDATE_ROLLBACK_FAILED)
+    newvalue(:UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS)
+    newvalue(:UPDATE_ROLLBACK_COMPLETE)
+    newvalue(:REVIEW_IN_PROGRESS)
+  end
+
+  newparam(:template_body) do
+    desc 'Structure containing the template body with a minimum length of 1 byte and a maximum length of 51,200 bytes. For more information, go to Template Anatomy in the AWS CloudFormation User Guide.'
+    validate do |value|
```

Comment:
Since the template will be JSON (correct?) then maybe it would be worth parsing the string here.